vochain/state.go

```diff
@@ -79,33 +83,61 @@ type ImmutableState struct {
 func NewState(dataDir string) (*State, error) {
 	var err error
 	vs := &State{}
-	//vs.Store = new(gravitonstate.GravitonState)
-	vs.Store = new(iavlstate.IavlState)
-
-	if err = vs.Store.Init(dataDir, "disk"); err != nil {
-		return nil, err
-	}
-
-	if err := vs.Store.AddTree(AppTree); err != nil {
-		return nil, err
-	}
-	if err := vs.Store.AddTree(ProcessTree); err != nil {
-		return nil, err
+	if err := initStore(dataDir, vs); err != nil {
+		return nil, fmt.Errorf("cannot init db: %s", err)
 	}
-	if err := vs.Store.AddTree(VoteTree); err != nil {
-		return nil, err
-	}
-
 	// Must be -1 in order to get the last committed block state, if not block replay will fail
 	if err = vs.Store.LoadVersion(-1); err != nil {
-		return nil, err
+		if err == iavl.ErrVersionDoesNotExist {
```

Comment:
Don't do that, its breaking the interface State. I think this error should be handled on the statedb/iavltree implementation, not here.